### PR TITLE
Option for serial recovery to respect "image" parameter of mcumgr image update

### PR DESCRIPTION
**Pull request by de-nordic**
The PR adds support for respecting the "image" parameter of the mcumgr image upload command in serial recovery.
This allows to directly upload image by number; the translation of the image number to the destination is done by platform specific implementation of the `flash_area_id_from_direct_image` function.

The PR contains two commits:
 - general code change, addition of `MCUBOOT_SERIAL_DIRECT_IMAGE_UPLOAD` that controls the options and prototype for `flash_area_id_from_direct_image`
 - Zephyr specific implementation.

### CODE REVIEW

**Comment by de-nordic** [comment]
FYI @sigvartmh

**Comment by de-nordic** [comment]
@hakonfam Can you look at this from the perspective of custom partition manager?

File: boot/zephyr/Kconfig [hunk 1]
```diff
@@ -493,6 +493,18 @@ config BOOT_SERIAL_CDC_ACM
...
 endchoice
 
+config MCUBOOT_SERIAL_DIRECT_IMAGE_UPLOAD
+	bool "Allow to select image number for DFU"
+	help
+	  With the option enabled, the mcuboot serial recovery will
+	  respect the "image" field in mcumgr image update frame
+	  header.
+	  The mapping of image number to partition is as follows:
+	    0 -> image-0 (slot 0)
```
Comment (nvlsianpu) [inline boot/zephyr/Kconfig:503]: I would use primary/secondary terminology as well:
0 -> image-0 (primary slot of the firs images pair)
1 -> image-0 (secondary slot of the firs images pair)

Comment (de-nordic) [reply boot/zephyr/Kconfig:503]: OK, will change.

**Review by nvlsianpu — APPROVED** [review]

**Comment by de-nordic** [comment]
Marked as DNM to change how images are mapped for Zephyr.

**Comment by de-nordic** [comment]
> 
> 
> Marked as DNM to change how images are mapped for Zephyr.

Changes applied, DNM removed.

**Review by d3zd3z — APPROVED** [review]

**Pull request merged**